corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -415,24 +427,41 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         },
 
         submitAction: function (e) {
+            var self = this;
             e.preventDefault();
-            FormplayerFrontend.trigger('clearNotifications', errorHTML, true);
 
             var invalidFields = [];
-            this.children.each(function (childView) {
-                if (!childView.isValid()) {
-                    invalidFields.push(childView.model.get('text'));
+            var errorHTML = gettext("Please check the following fields:");
+
+            FormplayerFrontend.trigger('clearNotifications', errorHTML, true);
+
+            var Util = hqImport("cloudcare/js/formplayer/utils/util");
+            var urlObject = Util.currentUrlToObject();
+            urlObject.setQueryData(self.getAnswers(), false);
+            var fetchingPrompts = FormplayerFrontend.getChannel().request("app:select:menus", urlObject);
+
+            $.when(fetchingPrompts).done(function (response) {
+                for (var i = 0; i < response.models.length; i++) {
+                    self.collection.models[i].set('error', response.models[i].get('error'));
                 }
-            });
+                self.children.each(function (childView) {
+                    if (!childView.isValid()) {
+                        invalidFields.push(childView.model.get('text'));
+                    }
+                });
 
-            if (invalidFields.length) {
-                var errorHTML = "Please enter values for the following fields:";
-                errorHTML += "<ul>" + _.map(invalidFields, function (f) { return "<li>" + f + "</li>"; }).join("") + "</ul>";
-                FormplayerFrontend.trigger('showError', errorHTML, true);
-                return;
-            }
+                if (invalidFields.length) {
+                    errorHTML += "<ul>" + _.map(invalidFields, function (f) {
+                        return "<li>" + DOMPurify.sanitize(f) + "</li>";
+                    }).join("") + "</ul>";
+                    FormplayerFrontend.trigger('showError', errorHTML, true);
```

Comment:
Noting that once https://github.com/dimagi/commcare-hq/pull/31922 is merged it would be good to not report these to sentry.

Reply:
This would end up being `FormplayerFrontend.trigger('showError', errorHTML, true, null, false);`, correct?

Reply:
yes, that should do it

Reply:
https://github.com/dimagi/commcare-hq/pull/31922/commits/6336777bb250404ba8e0c97b7670d7b0e4d877d3